piptools/utils.py

```diff
@@ -570,10 +570,21 @@ def override_defaults_from_config_file(
     else:
         config_file = Path(value)
 
-    config = parse_config_file(ctx, config_file)
-    if config:
-        _validate_config(ctx, config)
-        _assign_config_to_cli_context(ctx, config)
+    piptools_config, pipcompile_config, pipsync_config = parse_config_file(
```

Comment:
I don't think it's a good idea to return a tuple. If it is really necessary to have all these bits of data, wrap them into an object (dataclass?) and return that.

Though, I'd argue it should return a single config instead. Maybe, we need to pass an argument telling it for which command that config is to be extracted..

Reply:
Yes, I can move the test below to `parse_config_file()`.

Reply:
```suggestion
    config = Config.parse_file(config_file, ctx)
```

Given that you're in the process of refactoring this, how about introducing a `Config` class in `piptools/config.py` and moving the related logic there? This is an example of what the Config might look like:

```python
@dataclass
class Config:
    pip_tools: dict[str, Any]
    pip_compile: dict[str, Any]
    pip_sync: dict[str, Any]
    
    @classmethod
    def parse_file(filename):
        ...

    def validate(self):
        ...
        
    def normalize_keys(self):
        ...
    
    def invert_negative_bool_options(self, ctx):
        ...
        
    def get_default_map() -> dict[str, Any]:
        ...
     
    def assign_to_cli(ctx):
        ...    
```

Reply:
@atugushev What do you think, should we dedicate an own file to that?

Reply:
Sounds great 👍🏻

Reply:
@theryanwalker I like it, though I'd probably go for a different construction/classmethod name. How about `Config.from_file_path()`?